nixos/modules/services/torrent/transmission.nix

```diff
@@ -66,21 +67,15 @@ in
       };
 
       downloadDirPermissions = mkOption {
-        type = types.str;
+        type = types.nullOr types.str;
         default = "770";
         example = "775";
         description = ''
           The permissions to set for download-dir and incomplete-dir.
-          They will be applied on every service start.
+          If not null, they will be applied on every service start.
         '';
       };
 
-      port = mkOption {
```

Comment:
The RFC doesn't recommend removing commonly used options, right?

Reply:
@Ma27 You're right, and even more so since the RFC0042 states that "removing options has always been nigh impossible", but what should we do with this old ˋportˋ? Having both ˋportˋ (passed with ˋ--portˋ or with ˋconfig.services.transmission.settings.rpc-port = cfg.portˋ) and ˋrpc-portˋ (passed with ˋsettingsˋ) becomes a problem when considering which option should be used in a firewall config, because both can still be set to different values, even with ˋconfig.services.transmission.settings.rpc-port = cfg.portˋ. I could render it innocuous by adding an assertion ˋcfg.port == cfg.settings.rpc-portˋ but I would have to add a ˋ|| cfg.port == null || cfg.settings.rpc-port == nullˋ to handle the case where only one is defined, and then the firewall would have to deal with the null case... Adding all that complexity only to support a confusion that could simply be removed, feels wrong to me. ˋrpc-portˋ is not removable and is type-checkable with an assertion, so I think that removing ˋportˋ is the least confusing choice for the users interface and the less noisy one for the developers, unfortunately at the cost of backward-compatibility. This said, if you have a better idea, you're welcome :)

Reply:
Well, right now `rpc-port` is promoted as replacement for `port`. So why not setting `rpc-port` to `port` using `mkDefault` if `port` is declared?

Usually, the `settings` can be used to override values. See https://github.com/Infinisil/rfcs/blob/config-option/rfcs/0042-config-option.md#an-example for that.

Reply:
@Ma27, yes but knowing whether `port` is declared or not is the tricky part and why I was reluctant about using `null`, hopefully the same can be achieved with `options.services.transmission.port.isDefined` and removing `port`'s `default`, which is my current proposal. Note also that using `mkDefault` as in the RFC0042's example would not work because `settings` is currently of type `types.attrs` not `format.json.type`, hence `settings.rpc-port` is not actually a NixOS module option where `mkDefault` has a meaning.

Reply:
@Ma27 Is it possible to make the service deprecate the `port` option in favor of `settings.rpc-port`? I think I've seen that done with other services. Currently the mechanism that deals with both options seems to be broken.

Reply:
@doronbehar what's exactly the issue? The benefit of "non-settings" options is that you have documentation and explicit type-checking etc.

Reply:
@Ma27 maybe I misunderstood your discussion. I was trying to suggest a resolution to the issue of `port` vs `rpc-port`. But considering:

> The benefit of "non-settings" options is that you have documentation and explicit type-checking etc.

I'm not strongly opinionated.